src/Form/Template/LegacyFormSettingCompatibility.php

```diff
@@ -70,6 +70,15 @@ public function save( $formId, $settings ) {
 					continue;
 				}
 
+				/**
+				 * Prevents an undefined index error of the field id is not set in the settings group.
+				 * @link https://github.com/impress-org/givewp/pull/5882
+				 * @unreleased
+				 */
+				if ( ! isset( $settings[ $groupId ] ) || ! isset( $settings[ $groupId ][ $field['id'] ] ) ) {
```

Comment:
@kjohnson this function runs internally when you save the donation form on the form edit screen. If template-specific settings do not exist in `$_POST` then this function will not be called, so I am confused why this function is running in your case.

https://github.com/impress-org/givewp/blob/f9d39b1642a161d432c99729d107e87f45b209ca/includes/admin/forms/class-metabox-form-data.php#L66

Above filter cause of calling this function. I checked onboarding flow and I think the `$_POST[{templateId}]` setting will be empty, so we exit early in filter callback. Can you review it again?

Reply:
@ravinderk I think it is being triggered by a call to `\Give\Form\Template\Options::getCheckoutLabelField()`, see [P2P\DonationFormFactory](https://github.com/impress-org/give-peer-to-peer/blob/develop/src/P2P/Factories/DonationFormFactory.php#L53)

Reply:
@kbjohnson90 you are adding field registration array to setting value array.

Please replace
```php
TemplateOptions::getCheckoutLabelField()
```

with
```php
'checkout_label' => ewc_html__( 'Donate Now', 'give-p2p ')
```

![Edit Donation Form ‹ GiveWP — WordPress 2021-07-16 at 10 22 33 PM](https://user-images.githubusercontent.com/1784821/125982362-40f652ff-45a5-40ae-8320-8ccb148a2b16.jpg)

Reply:
Implemented in P2P.